string[] detected_token
bool picked
bool placed
bool moved_back
